### PR TITLE
Assignment 10

### PR DESCRIPTION
**Pull request by ap221882**
# Added Dark-Light Theme using React's context API

https://user-images.githubusercontent.com/60094012/208234947-ba0ec0c2-4306-4fdf-974b-97a85a52a6e1.mov

### CODE REVIEW

File: Day7n8n9n10/github-api-project-ts/src/containers/Navigator/Navigator.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+import { NavigationButton } from "../../components";
+import classes from "./navigator.styles.module.css";
+
+const Navigator = () => {
+  return (
+    <div className={classes.navigator__container}>
+      <NavigationButton buttonText='Show Avengers Team' navigateTo='/members' />
```
Comment (nitishnivedan) [inline Day7n8n9n10/github-api-project-ts/src/containers/Navigator/Navigator.tsx:9]: can use an array of object having keys as buttonText and navigateTo. And can use a map for this

File: Day7n8n9n10/github-api-project-ts/src/containers/Header/Header.tsx [hunk 2]
```diff
@@ -2,12 +2,14 @@ import React from "react";
 
 import classes from "./header.styles.module.css";
 import ClassBasedComponent1 from "../../miscellaneous/ParentClassBasedComponent";
+import { useThemeContext } from "../../contexts/ThemeContext";
 
 type Props = {};
 
 const Header = (props: Props) => {
+  const {mode}=useThemeContext()
```
Comment (nitishnivedan) [inline Day7n8n9n10/github-api-project-ts/src/containers/Header/Header.tsx:10]: i was using in my code, something like this  const { color } = useContext(ThemeContext);  //was passing the Themecontext to useContext,
 does  **const {mode}=useThemeContext()**  -> also do the same thing?

File: Day7n8n9n10/github-api-project-ts/src/containers/SearchGithubUsers/SearchGithubUsers.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import data from "../../utils/state-city-data.json";
+import classes from "./searchGithubUsers.styles.module.css";
+import Header from "./SearchGithubUsersUI/Heading";
+
+const SearchGithubUsers = () => {
+  const [states, setStates] = useState<Array<string>>([]);
+  const [selectedState, setSelectedState] = useState<string>(
+    "Andaman and Nicobar Islands"
+  );
+  const [selectedCity, setSelectedCity] = useState<string>("Gurugram");
```
Comment (nitishnivedan) [inline Day7n8n9n10/github-api-project-ts/src/containers/SearchGithubUsers/SearchGithubUsers.tsx:16]: better not to hardcode  the city. can pass as an empty string and handle the empty string if it is breaking the page somewhere.

File: Day7n8n9n10/github-api-project-ts/src/containers/Header/Header.tsx (hunk 2 above)
Comment (ap221882) [reply Day7n8n9n10/github-api-project-ts/src/containers/Header/Header.tsx:10]: It's the same but is useful when you're using global contexts, you'll reduce one less import and don't have to write everytime ```const { color } = useContext(ThemeContext)```.
I've exported it like `export const useThemeContext = useContext(ThemeContext)`

File: Day7n8n9n10/github-api-project-ts/src/containers/SearchGithubUsers/SearchGithubUsers.tsx (hunk 3 above)
Comment (ap221882) [reply Day7n8n9n10/github-api-project-ts/src/containers/SearchGithubUsers/SearchGithubUsers.tsx:16]: Yes, nice suggestion. Will fix it.

File: Day7n8n9n10/github-api-project-ts/src/containers/Navigator/Navigator.tsx (hunk 1 above)
Comment (ap221882) [reply Day7n8n9n10/github-api-project-ts/src/containers/Navigator/Navigator.tsx:9]: Yes, but I thought that that would also be hard coding only. Let's do this your suggested way and see how it is nice to have.

File: Day7n8n9n10/github-api-project-ts/src/containers/NavBar/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+
```
Comment (nam334) [inline Day7n8n9n10/github-api-project-ts/src/containers/NavBar/Navbar.tsx:2]: u can avoid importing react as it is no longer needed in jsx components

**Comment by roshantrivedi** [comment]
Wow.. looks cool!

**Pull request merged**